static/app/components/forms/selectAsyncField.tsx

```diff
@@ -10,7 +10,13 @@ import {GeneralSelectValue} from 'sentry/components/forms/selectControl';
 
 export interface SelectAsyncFieldProps
   extends Omit<InputFieldProps, 'highlighted' | 'visible' | 'required' | 'value'>,
-    SelectAsyncControlProps {}
+    SelectAsyncControlProps {
+  /**
+   * Similar to onChange, except it provides the entire option object (including label) when a
+   * change is made to the field. Occurs after onChange.
+   */
+  onChangeOption?: (option: GeneralSelectValue, event: any) => void;
```

Comment:
This was the most managable way I figured to bubble up the `label` information to the form component. The FormModel does not contain information about the labels in an accessible way. It would not be performant to iterate over all options searching for a shared key. 

Instead, this function will only fire when a selection is made so the parent form only see's labels uses labels that will be submitted (instead of iterating over all options).